### ISSUE TITLE
Remove old method signature

### ISSUE THREAD

**Issue by dahong67** [dahong67/GCPDecompositions.jl#23]
One more TODO remaining from #22:
https://github.com/dahong67/GCPDecompositions.jl/blob/4b5431a4248ef9b50dbe20b9fafb0f16350246cc/src/gcp-opt.jl#L48-L55

Will require reworking how we do the testing since we currently use that signature. For example:
https://github.com/dahong67/GCPDecompositions.jl/blob/4b5431a4248ef9b50dbe20b9fafb0f16350246cc/test/items/gcp-opt.jl#L51-L58

Since this would affect ongoing work in #20, can address this in a separate PR after #20 is merged. Should probably rethink how we do this testing anyway.

_Originally posted by @dahong67 in https://github.com/dahong67/GCPDecompositions.jl/issues/22#issuecomment-1760523386_
